### ISSUE TITLE
rdf-ext does not contain a package.json file

### ISSUE THREAD

**Issue by tim-tx** [SynBioDex/sboljs#45]
$ mkdir tmp
    $ npm install sboljs
    $ rm -rf node_modules
    $ npm install
    npm ERR! code ENOLOCAL
    npm ERR! Could not install from "node_modules/sboljs/rdf-ext@git:/github.com/rdf-ext/rdf-ext.git#a01be12f62a55182edfffc3c81ff37befcbc85b8" as it does not contain a package.json file.

    npm ERR! A complete log of this run can be found in:
    npm ERR!     /home/me/.npm/_logs/2019-03-14T14_28_30_904Z-debug.log

Instead one has to do `npm install sboljs` again and then `npm install`.

**Comment by dany-fu** [comment]
@cjmyers we are getting this too, on travis, but not locally

**Comment by cjmyers** [comment]
@udp Any ideas on this?  @dany-fu does running install twice help?  Are you definitely getting the latest version of sboljs which should be 2.2.2.

**Comment by dany-fu** [comment]
Yes we're using the latest, our CI/CD pipeline errors out when we push to Git so there's not an option to run install twice. It's been building just fine until today, including a PR two days ago on v2.1.13, and I would have thought it was a bug on our end had we not seen this issue.

**Comment by cjmyers** [comment]
rdf-ext can be found here:

https://github.com/rdf-ext/rdf-ext

It is published on npm here:

https://www.npmjs.com/package/rdf-ext

It has not been updated in 7 months and it definitely has a package.json

**Comment by cjmyers** [comment]
We are actually pinning to a particular commit:

https://github.com/rdf-ext/rdf-ext#a01be12f62a55182edfffc3c81ff37befcbc85b8

I seem to recall that @udp modified rdf-ext in some way, which is why we are pinned to a commit.  I wonder if this is the issue, since it is not simply a pull from npm.

**Comment by jamesamcl** [comment]
No idea why this would be - maybe an npm bug.  I will investigate.

What version of npm are you using?

**Comment by dany-fu** [comment]
@udp @cjmyers I compared the package-lock.json files from a passing branch and then manually updated these two lines from 

broken:
```
"rdf-ext": "rdf-ext@git://github.com/rdf-ext/rdf-ext.git#a01be12f62a55182edfffc3c81ff37befcbc85b8",
"rdf-graph-array": "git+https://github.com/udp/rdf-graph-array.git",
```

to passing:
```
"rdf-ext": "git://github.com/rdf-ext/rdf-ext.git#a01be12f62a55182edfffc3c81ff37befcbc85b8",
"rdf-graph-array": "git+https://github.com/udp/rdf-graph-array.git#98b1a0c674ed2b07993d54ab7d2a3b84e661bc3e",
``` 

I think the issue was with rdf-graph-array

**Comment by cjmyers** [comment]
@udp didn't you need to modify rdf-graph-array for some reason?  

@dany-fu are things working again for you then?

**Comment by dany-fu** [comment]
Yes, travis is passing now, but it means I have to manually change these lines every time package-lock updates

**Comment by cjmyers** [comment]
Hmm, @udp can't we somehow update package.json, so this update of this package won't happen?

**Comment by jamesamcl** [comment]
Please can you try with sboljs 2.2.3? (now published on npm)

**Comment by dany-fu** [comment]
```
$ npm install
npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/sboljs/rdf-graph-array@git+https:/github.com/udp/rdf-graph-array.git#98b1a0c674ed2b07993d54ab7d2a3b84e661bc3e" as it does not contain a package.json file.
```

**Comment by cjmyers** [comment]
Hmm, that is strange.  I looked at the commits by @udp and it seems he just added to package.json what you needed to add to package-lock.json.  Do you see any differences in what he has done to what you did?

**Comment by dany-fu** [comment]
After updating to 2.2.3 `package-lock.json` has these lines which fail with the error above

```
"rdf-ext": "git+https://github.com/rdf-ext/rdf-ext.git#a01be12f62a55182edfffc3c81ff37befcbc85b8",
"rdf-graph-array": "rdf-graph-array@git+https://github.com/udp/rdf-graph-array.git#98b1a0c674ed2b07993d54ab7d2a3b84e661bc3e",
```

I manually change these lines to what I had before
```
"rdf-ext": "git://github.com/rdf-ext/rdf-ext.git#a01be12f62a55182edfffc3c81ff37befcbc85b8",
"rdf-graph-array": "git+https://github.com/udp/rdf-graph-array.git#98b1a0c674ed2b07993d54ab7d2a3b84e661bc3e",
```

They are both pinned to the same commits but the prefixes are different. I don't know enough about npm to understand why the latter passes on Travis but not the former

**Comment by cjmyers** [comment]
@udp any ideas?

**Comment by jamesamcl** [comment]
Please try again with 2.2.4
